source/val/validate_debug.cpp

```diff
@@ -51,9 +53,33 @@ spv_result_t ValidateLine(ValidationState_t& _, const Instruction* inst) {
   return SPV_SUCCESS;
 }
 
+bool IsDebugInstruction(const Instruction* inst) {
+  switch (inst->opcode()) {
+    case SpvOpName:
+    case SpvOpMemberName:
+    case SpvOpSource:
+    case SpvOpSourceContinued:
+    case SpvOpSourceExtension:
+    case SpvOpString:
+    case SpvOpLine:
+    case SpvOpNoLine:
```

Comment:
@alan-baker do we need to worry about things like OpDecorateStringGOOGLE here?

Reply:
Those are decorations not debug instructions. So it would depend on what the env spec says for supported decorations.

Reply:
The spec is defining a whitelist of decorations, so likely they will be handled by the default if not on this list, then not allowed. If/when we need to deal with these decorations they will need to be handled by something different then this opcode check.